test/Spec.hs

```diff
@@ -1,3 +1,104 @@
+{-# language PartialTypeSignatures #-}
+-- We definitely don't want to infer the types of the test functions
+-- based on how they're *used*; if their implementations and (partial)
+-- signatures aren't sufficient, then we don't want them to have
+-- Typeable instances.
+{-# language NoMonomorphismRestriction #-}
+{-# OPTIONS_GHC -Wno-partial-type-signatures #-}
+
+module Main (main) where
+import Data.Monoid (Sum (..), All (..))
+import Control.Newtype.Generic
+import Type.Reflection
+
+-- We don't want defaults making it look like things are working
+-- if they're not!
+default ()
```

Comment:
Interesting! What does this do?

Reply:
I believe this prevents GHC from defaulting things to Integer/Float/whatever. See https://wiki.haskell.org/Keywords#default. We don't want any defaulting going on, just in case GHC can't infer the type of something